### PR TITLE
[MRG] Convert docstring of Axes.pie() into numpydoc 

### PR DESCRIPTION
**Pull request by kahnchana**
#### What does this implement/fix? Explain your changes.
The docstring of the function pie() in lib/matplotlib/axes/_axes.py has been converted to numpydoc format. Also, pep-8 style errors in the code have been fixed.

### CODE REVIEW

**Comment by QuLogic** [comment]
There's a lot of extraneous whitespace changes here. While correcting this is not unwelcome, it makes it rather difficult to say what parts of this change are actually relevant. It would be better to have a separate PR for whitespace correction, or only do it on the parts that you are actually editing.

Also, please write actual commit message, not just a single dot.

**Comment by kahnchana** [comment]
Added the changes you suggested. I removed all the whitespace fixing. Will create a separate issue for that later. And added the exact commit name. Sorry about previous mistake.

**Comment by QuLogic** [comment]
Hmm, I should have been a bit clearer. It's not good to revert this because now there's a commit making the whitespace changes and another commit undoing those, and then if you open a separate PR, there'll be another commit fixing the whitespace again.

If you know how to rebase, it would be a little better if you rebased and squashed these commits all into a single one.

File: lib/matplotlib/axes/_axes.py [hunk 1]
```diff
@@ -2475,99 +2475,102 @@ def pie(self, x, explode=None, labels=None, colors=None,
...
         by default starting from the x-axis.
 
-        Keyword arguments:
+        Parameters
+        ----------
+        x : array
+            The input array used to make the pie chart.
 
-          *explode*: [ *None* | len(x) sequence ]
+        explode: None or array
```
Comment (QuLogic) [inline lib/matplotlib/axes/_axes.py]: Add a space before the colon for all parameters.

File: lib/matplotlib/axes/_axes.py [hunk 2]
```diff
@@ -2475,99 +2475,102 @@ def pie(self, x, explode=None, labels=None, colors=None,
...
-          *labeldistance*: scalar
+        labeldistance: float
             The radial distance at which the pie labels are drawn
 
-          *shadow*: [ *False* | *True* ]
+        shadow: bool
             Draw a shadow beneath the pie.
 
-          *startangle*: [ *None* | Offset angle ]
+        startangle: None or Offset angle
```
Comment (QuLogic) [inline lib/matplotlib/axes/_axes.py]: `Offset angle` is not a type; should be `float`.

File: lib/matplotlib/axes/_axes.py [hunk 3]
```diff
@@ -2475,99 +2475,102 @@ def pie(self, x, explode=None, labels=None, colors=None,
...
-          *textprops*: [ *None* | dict of key value pairs ]
+        textprops: None or dict
             Dict of arguments to pass to the text objects.
 
-          *center*: [ (0,0) | sequence of 2 scalars ]
-          Center position of the chart.
-
-          *frame*: [ *False* | *True* ]
-            Plot axes frame with the chart.
+        center:  list of int
```
Comment (QuLogic) [inline lib/matplotlib/axes/_axes.py]: Extra space. Also, why `int`?

File: lib/matplotlib/axes/_axes.py [hunk 4]
```diff
@@ -2475,99 +2475,102 @@ def pie(self, x, explode=None, labels=None, colors=None,
...
-        The pie chart will probably look best if the figure and axes are
-        square, or the Axes aspect is equal.  e.g.::
+        rotatelabels: bool
+            Rotate each label to the angle of the corresponding slice if true.
 
-          figure(figsize=(8,8))
-          ax = axes([0.1, 0.1, 0.8, 0.8])
+        Returns
+        -------
+        patches: list
```
Comment (QuLogic) [inline lib/matplotlib/axes/_axes.py]: Also add space before colon of every return value.

File: lib/matplotlib/axes/_axes.py [hunk 5]
```diff
@@ -2475,99 +2475,102 @@ def pie(self, x, explode=None, labels=None, colors=None,
...
+            A sequence of :class:`matplotlib.patches.Wedge` instances
 
-        or::
+        texts: list
+            A is a list of the label :class:`matplotlib.text.Text` instances.
 
-          axes(aspect=1)
+        autotexts: list
+            A is a list of :class:`~matplotlib.text.Text` instances for the
+            numeric labels. Is returned only is if parameter *autopct* is
```
Comment (QuLogic) [inline lib/matplotlib/axes/_axes.py]: is if -> if

File: lib/matplotlib/axes/_axes.py [hunk 6]
```diff
@@ -2475,99 +2475,102 @@ def pie(self, x, explode=None, labels=None, colors=None,
...
+            not *None*. If *autopct* is *None*, the tuple returned
+            is (patches, texts).
```
Comment (QuLogic) [inline lib/matplotlib/axes/_axes.py]: Leave out this last sentence.

File: lib/matplotlib/axes/_axes.py (hunk 3 above)
Comment (kahnchana) [inline lib/matplotlib/axes/_axes.py]: My bad. Changing to float.

**Comment by kahnchana** [comment]
I tried rebasing and made a mistake. I ended up adding additional commits. I am trying to fix this and figure out the rebasing. 

Also made the changes suggested.

**Comment by kahnchana** [comment]
Shall I remove this PR and open a new PR with the changes you suggested?

**Comment by dstansby** [comment]
It looks like quite a few things have gone wrong with git here; if it would be easier for you to open a new PR, please feel free!

**Comment by phobson** [comment]
@kahnchana I have cleaned up the commits to this branch locally.

I'm closing this in favor of #8371

**Pull request closed without merging**